### DISCUSSION TITLE
"number" still not in spacy's vocab

### DISCUSSION THREAD

**Discussion by minimario** [explosion/spaCy#10123] [Help: Other Questions] [answered]
There's a relevant issue here, but "number" doesn't seem to be in spacy's vocab? https://github.com/explosion/spaCy/issues/2561

## How to reproduce the behaviour
```
def is_word_in_nlp_vocab(word: str, nlp) -> bool:
    return word in list(nlp.vocab.strings)
nlp2 = spacy.load('en_core_web_lg')
is_word_in_nlp_vocab("number", nlp2)
False
```

## Your Environment
- **spaCy version:** 3.2.1
- **Platform:** Linux-5.11.0-46-generic-x86_64-with-glibc2.31
- **Python version:** 3.9.7
- **Pipelines:** en_core_web_lg (3.2.0), en_core_web_sm (3.2.0), en_core_web_md (3.2.0)

**Comment by polm** [comment, marked as answer]
The particular way you're checking this returning false is a quirk of the implementation of the string store. "number" is one of the strings stored as a [symbol](https://github.com/explosion/spaCy/blob/master/spacy/symbols.pyx), which means it is stored in a special way and won't come up if you iterate over the keys in the string store. On the other hand this is True:

```
import spacy

nlp = spacy.load('en_core_web_lg')

print("number" in nlp.vocab.strings) # True
```

I didn't address this last time because the issue just seemed resolved either way, but can you clarify why this is a problem for you? The string store is an implementation detail and the presence or absence of strings in it is basically arbitrary - it shouldn't be used to check if something is OOV or anything.

**Reply by minimario** [reply]
Thanks, that makes sense :)

So my use case is that I have a set of words, either fairly common words or garbles phrases that aren't words and want to separate the two... I thought this was the right way to do that but supposedly not. Should I be using is_oov instead?

**Reply by adrianeboyd** [reply]
No, there aren't any word lists in the provided trained pipelines that are suitable for spell checking or similar tasks. `is_oov` is used to check whether there's a word vector (for v2.3+; note that earlier versions had a slightly different use), but the list of tokens with vectors is too large and noisy to be useful for spell checking. I mean, I suppose it wouldn't hurt for you to try and see if it did perform well enough, but it's definitely not intended for this kind of task.
